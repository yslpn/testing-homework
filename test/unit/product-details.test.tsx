import React from "react";

import { render, screen } from "../test-utils";
import { ProductDetails } from "../../src/client/components/ProductDetails";

const products = [
  {
    id: 0,
    name: "Refined Sausages",
    description:
      "The beautiful range of Apple Naturalé that has an exciting mix of natural ingredients. With the Goodness of 100% Natural Ingredients",
    price: 347,
    color: "fuchsia",
    material: "Granite",
  },
  {
    id: 1,
    name: "Rustic Sausages",
    description:
      "Boston's most advanced compression wear technology increases muscle oxygenation, stabilizes active muscles",
    price: 445,
    color: "white",
    material: "Fresh",
  },
  {
    id: 2,
    name: "Incredible Bacon",
    description: "The Football Is Good For Training And Recreational Purposes",
    price: 972,
    color: "ivory",
    material: "Frozen",
  },
];

test("Страница продукта отрендерилась успешно", () => {
  const { container } = render(<ProductDetails product={products[0]} />, {
    preloadedState: { cart: [], details: [] },
  });

  expect(container).toMatchSnapshot();

  // fireEvent.change(input, { target: { value: "новое значение" } });

  // await screen.goto("http://localhost:3000/hw/store/cart");
  // await screen.locator("#f-name").fill("Ivan Ivanov");
  // await screen.locator("#f-phone").fill("+995591805020");
  // await screen.locator("#f-address").fill("Tbilisi, 40 Simon Chikovani Street");

  // await screen.locator(".Form-Submit").click();
});
