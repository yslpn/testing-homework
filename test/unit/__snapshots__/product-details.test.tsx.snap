// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Страница продукта отрендерилась успешно 1`] = `
<div>
  <div
    class="ProductDetails row"
  >
    <div
      class="col-12 col-sm-5 col-lg-4"
    >
      <img
        class="Image"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkMAYAADkANVKH3ScAAAAASUVORK5CYII="
      />
    </div>
    <div
      class="col-12 col-sm-7 col-lg-6"
    >
      <h1
        class="ProductDetails-Name"
      >
        Refined Sausages
      </h1>
      <p
        class="ProductDetails-Description"
      >
        The beautiful range of Apple Naturalé that has an exciting mix of natural ingredients. With the Goodness of 100% Natural Ingredients
      </p>
      <p
        class="ProductDetails-Price fs-3"
      >
        $
        347
      </p>
      <p>
        <button
          class="ProductDetails-AddToCart btn btn-primary btn-lg"
        >
          Add to Cart
        </button>
      </p>
      <dl>
        <dt>
          Color
        </dt>
        <dd
          class="ProductDetails-Color text-capitalize"
        >
          fuchsia
        </dd>
        <dt>
          Material
        </dt>
        <dd
          class="ProductDetails-Material text-capitalize"
        >
          Granite
        </dd>
      </dl>
    </div>
  </div>
</div>
`;
